pkg/blobstore/cloud/cloud_blob_access.go

```diff
@@ -77,16 +88,31 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 
 func (ba *cloudBlobAccess) FindMissing(ctx context.Context, digests digest.Set) (digest.Set, error) {
 	missing := digest.NewSetBuilder()
+
 	for _, blobDigest := range digests.Items() {
-		if exists, err := ba.bucket.Exists(ctx, ba.getKey(blobDigest)); err != nil {
-			return digest.EmptySet, err
-		} else if !exists {
+		err := ba.touchBlob(ctx, blobDigest)
+		switch gcerrors.Code(err) {
+		case gcerrors.OK:
+			// Not missing
+		case gcerrors.NotFound:
+			// Missing
 			missing.Add(blobDigest)
+		default:
+			return digest.EmptySet, err
 		}
 	}
+
 	return missing.Build(), nil
 }
 
 func (ba *cloudBlobAccess) getKey(digest digest.Digest) string {
 	return ba.keyPrefix + digest.GetKey(ba.digestKeyFormat)
 }
+
+func (ba *cloudBlobAccess) touchBlob(ctx context.Context, blobDigest digest.Digest) error {
```

Comment:
Nit: If you change this function to take a `key string` instead of a `blobDigest digest.Digest`, `ba.getKey()` gets called less frequently.

Reply:
I believe this evaporates when compiled anyway, but sure, it makes little difference